src/Spice86.Core/Emulator/VM/Machine.cs

```diff
@@ -190,12 +184,11 @@ public class Machine : IDisposable {
             serviceProvider.GetService<ILoggerService>(),
             TimerInt8Handler);
         Register(SystemClockInt1AHandler);
-        DosInt20Handler = new DosInt20Handler(this, serviceProvider.GetService<ILoggerService>());
-        Register(DosInt20Handler);
-        DosInt21Handler = new DosInt21Handler(this, serviceProvider.GetService<ILoggerService>());
-        Register(DosInt21Handler);
-        DosInt2fHandler = new DosInt2fHandler(this, serviceProvider.GetService<ILoggerService>());
-        Register(DosInt2fHandler);
+
+        // Initialize DOS.
+        Dos = new Dos(this, new LoggerService(), configuration);
```

Comment:
serviceProvider.GetService<ILoggerService>(), as it is a Singleton service.

Reply:
But good catch ! I should probably make its ctor internal to the Spice86.Logging assembly.

Reply:
done